agent/internal/taskoutput/test_log.go

```diff
@@ -0,0 +1,299 @@
+
+			h.followFileCount++
+			h.wg.Add(1)
+			go h.followFile(workerCtx, event)
+		case err := <-h.watcher.Error:
+			h.logger.Execution().Critical(errors.Wrap(err, "watching test log directory"))
+			return
+		case <-ctx.Done():
+			h.logger.Execution().Warning("context canceled, exiting test log directory watcher")
+			return
+		case <-h.watcher.Closed:
+			return
+		}
+	}
+}
+
+func (h *testLogDirectoryHandler) getSpecFile() {
+	h.logger.Task().Infof("detected first test log file, getting test log spec file")
+
+	data, err := os.ReadFile(filepath.Join(h.dir, testLogSpecFilename))
+	if err != nil {
+		h.logger.Task().Warning(errors.Wrap(err, "reading test log spec; falling back to default spec"))
+		return
+	}
+	if err = yaml.Unmarshal(data, &h.spec); err != nil {
+		h.logger.Task().Warning(errors.Wrap(err, "unmarshalling test log spec; falling back to default spec"))
+		return
+	}
+
+	if err = h.spec.Format.validate(); err != nil {
+		h.logger.Task().Warning(errors.Wrapf(err, "invalid test log format specified; falling back to default text format"))
+	}
+}
+
+func (h *testLogDirectoryHandler) followFile(ctx context.Context, event watcher.Event) {
+	defer func() {
+		h.logger.Task().Critical(recovery.HandlePanicWithError(recover(), nil, "test log file follower"))
+	}()
+	defer h.wg.Done()
+
+	h.once.Do(h.getSpecFile)
+
+	h.logger.Task().Infof("new test log file '%s' found, initiating automated ingestion", event.Path)
+
+	logPath, err := filepath.Rel(h.dir, event.Path)
+	if err != nil {
+		h.logger.Task().Error(errors.Wrapf(err, "getting relative path for test log file '%s'", event.Path))
+		return
+	}
+
+	t, err := follower.New(event.Path, follower.Config{
+		Whence: io.SeekStart,
+		Offset: 0,
+		Reopen: true,
+	})
+	if err != nil {
+		h.logger.Task().Error(errors.Wrapf(err, "creating follower for test log file '%s'", event.Path))
+		return
+	}
+	defer t.Close()
+
+	sender, err := h.createSender(ctx, logPath)
+	if err != nil {
+		h.logger.Task().Error(errors.Wrapf(err, "creating Sender for test log '%s'", event.Path))
+		return
+	}
+	defer func() {
+		if err = t.Err(); err != nil {
+			h.logger.Task().Error(errors.Wrapf(err, "following test log file '%s'", event.Path))
+		}
+		if err = sender.Close(); err != nil {
+			h.logger.Task().Error(errors.Wrapf(err, "closing Sender for test log '%s'", event.Path))
+		}
+	}()
+
+	lines := t.Lines()
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case line, ok := <-lines:
+			if !ok {
+				return
+			}
+			sender.Send(message.NewDefaultMessage(level.Info, line.String()))
+		}
+	}
+}
+
+func (h *testLogDirectoryHandler) close(_ context.Context) error {
+	h.watcher.Close()
+	h.wg.Wait()
+
+	return nil
+}
+
+type testLogSpec struct {
+	SchemaVersion string        `yaml:"schema_version"`
+	Format        testLogFormat `yaml:"format"`
+}
+
+const testLogSpecFilename = "log_spec.yaml"
+
+func (s testLogSpec) getParser() taskoutput.LogLineParser {
+	switch s.Format {
+	case testLogFormatTextTimestamp:
```

Comment:
is this the resmoke format?

Reply:
Yes, this is the format that resmoke will be writing log lines in.

We are starting with support for two formats: 
- `text`: lines are appended as is and timestamps are added upon ingestion by the agent
- `text-timestamp`: lines should be prefixed with the nanosecond UNIX timestamp and are parsed accordingly upon ingestion

You can imagine that we expand these to include priority and other types of accepted formats.